core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -66,8 +66,8 @@ public abstract class LazyBuildMixIn<JobT extends Job<JobT, RunT> & Queue.Task &
 
     private static final Logger LOGGER = Logger.getLogger(LazyBuildMixIn.class.getName());
 
-    @SuppressWarnings("deprecation") // [JENKINS-15156] builds accessed before onLoad or onCreatedFromScratch called
-    private @NonNull RunMap<RunT> builds = new RunMap<>();
+    // [JENKINS-15156] builds accessed before onLoad or onCreatedFromScratch called
+    private @NonNull RunMap<RunT> builds = new RunMap<>(asJob());
```

Comment:
Had to rearrange the order of initialization a bit so that https://github.com/jenkinsci/jenkins/blob/905844f16ba9deac3c25b2f7c2a811bbfa41990d/core/src/main/java/jenkins/model/lazy/AbstractLazyLoadRunMap.java#L339-L340 would have a `Job` available for the listener when first loading the build list.